packages/cxl-ui/src/components/cxl-inline-comment-context-menu.js

```diff
@@ -0,0 +1,77 @@
+import { customElement } from 'lit-element';
+import '@conversionxl/cxl-lumo-styles';
+import { ContextMenuElement } from '@vaadin/vaadin-context-menu/src/vaadin-context-menu.js';
+
+@customElement('cxl-inline-comment-context-menu')
+export class CXLInlineCommentContextMenu extends ContextMenuElement {
+  ready() {
+    super.ready();
+    this.appendContextMenuScript();
+  }
+
+  static get is() {
+    return 'cxl-inline-comment-context-menu';
+  }
+
+  static get properties() {
+    return {
+      ...super.properties,
+      textareaPlaceholder: {
+        type: String,
+        value: 'Put your comment here',
+        notify: true,
+        reflectToAttribute: true
+      },
+      buttonText: {
+        type: String,
+        value: 'Save comment',
+        notify: true,
+        reflectToAttribute: true
+      },
+      buttonTextSavingComment: {
+        type: String,
+        value: 'Saving...',
+        notify: true,
+        reflectToAttribute: true
+      }
+    };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  appendContextMenuScript() {
+    // eslint-disable-next-line func-names
+    const doc = window.document;
+    const contextMenu = this;
+    contextMenu.closeOn = 'blur';
+    contextMenu.renderer = root => {
+      let listBox = root.firstElementChild;
+      // Check if there is a list-box generated with the previous renderer call to update its content instead of recreation
+      if (listBox) {
+        listBox.innerHTML = '';
+      } else {
+        listBox = doc.createElement('vaadin-list-box');
+        root.appendChild(listBox);
+      }
+      const txt = doc.createElement('textarea');
+      txt.placeholder = this.textareaPlaceholder;
+      listBox.appendChild(txt);
+      const but = doc.createElement('button');
+      but.textContent = this.buttonText;
+      but.onclick = () => {
+        but.disabled = true;
+        but.textContent = this.buttonTextSavingComment;
+        const evt = new CustomEvent('cxl-save-inline-comment', {
+          bubbles: true,
+          composed: true,
+          detail: {
+            comment: txt.value,
+            context: contextMenu.innerHTML,
+            elementId: this.getAttribute('id')
+          }
+        });
+        contextMenu.dispatchEvent(evt);
+      };
+      listBox.appendChild(but);
+    };
+  }
```

Comment:
I gotta refresh my memory on context menu population options - this code currently looks very noisy and manual hand-holding. Wondering if there's a better way :thinking:

Reply:
Is it not possible to use `<template>`-based render here?

Reply:
Please try `<template>` for context menu form.

Clarity becomes massively better. JS-based element creation is such messy code soup, it should be the last resort where possible.

```
Polymer Templates

Alternatively to using the renderer, you can populate the menu content using Polymer's Templates:

<vaadin-context-menu>
  <template>
    <vaadin-list-box>
      <vaadin-item>First menu item</vaadin-item>
      <vaadin-item>Second menu item</vaadin-item>
    </vaadin-list-box>
  </template>
</vaadin-context-menu>
```

https://cdn.vaadin.com/vaadin-context-menu/5.0.0-alpha2/#/elements/vaadin-context-menu

Reply:
@lkraav I've send the latest commit with refactoring proposition. Unfortunately because `vaadin-context-menu` is directly extended from the `ContextMenuElement` class, I had to try to override `static get template()` method, but this one is working as expected, because for some unknown to me reason - when I re-define this method in the child class, I cannot query them via `root.querySelector()` and the only element that is accessible is the parent's `<vaadin-context-menu-overlay id="overlay">` element. Further more - to do that I must add Polymer dependency to the component, just to access old `html` tag function, because when I try to use `html` tag function from LitElement library it doesn't work.
I would be grateful if you could provide any example of how to override the Vaadin components templates and query elements in the overridden templates. I tried it also using `this.$.elementId` property like they do it in Vaadin Components, but without success.

Reply:
OK this needs looking into.

I'll try to get to it within this week.